python/mxnet/gluon/parameter.py

```diff
@@ -904,7 +904,11 @@ def zero_grad(self):
             return
 
         for arr in arrays.values():
-            mx.nd.reset_arrays(*arr, num_arrays=len(arr))
+            if is_np_array():
+                for ele in arr:
+                    ele[:] = 0
```

Comment:
Need to use `ele[()] = 0` here for supporting zero-dim ndarrays as well. `slice(None)` is not allowed as an index for those ndarrays.

Reply:
Nice catch! I was not aware of that.